<div style="background:linear-gradient(135deg,#1e1b4b 0%,#4338ca 55%,#6366f1 100%);border-radius:18px;padding:34px 30px;color:#fff;font-family:Inter,Segoe UI,sans-serif">
  <div style="font-size:12px;letter-spacing:3px;color:#c7d2fe;font-weight:700;text-transform:uppercase">Chapter 31 · Solutions</div>
  <div style="font-size:34px;font-weight:900;line-height:1.1;margin:10px 0 6px">Rules Challenges, Worked Answers ✅</div>
  <div style="font-size:15px;color:#e0e7ff;max-width:720px;line-height:1.6">Full solutions to the five Rules of Probability challenges. Try them yourself first, then compare.</div>
  <div style="margin-top:16px;font-size:13px;color:#cbd5e1">Statistics, Data Science and AI: A Visual Handbook · John Fisher · 2026</div>
</div>

## ⚙️ Setup

In [1]:
import numpy as np
rng = np.random.default_rng(313)
print("ready")

ready


<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#4f46e5;letter-spacing:1px">CHALLENGE 1 · ADDITION WITH OVERLAP</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Count each card once</div>
<div style="color:#4a5578;margin-top:6px">Draw one card. Find P(red OR king). Identify the overlap and apply the addition rule, then verify by simulation.</div>
</div>

In [2]:
p_red = 26/52; p_king = 4/52; p_red_king = 2/52   # king of hearts, king of diamonds
p_or = p_red + p_king - p_red_king
print(f"P(red OR king) = 26/52 + 4/52 - 2/52 = {p_or:.3f}")
# simulate: encode a deck
colors = np.repeat([0,1],26)            # 0=black,1=red
ranks  = np.tile(np.arange(13),4)       # 12 = king
idx = rng.integers(0,52,300_000)
emp = np.mean((colors[idx]==1) | (ranks[idx]==12))
print(f"empirical (300k draws)             = {emp:.3f}")

P(red OR king) = 26/52 + 4/52 - 2/52 = 0.538
empirical (300k draws)             = 0.539


**Answer:** there are 26 red cards and 4 kings, but 2 cards (king of hearts, king of diamonds) are both, so they would be counted twice. P(red or king) = 26/52 + 4/52 - 2/52 = 28/52 ≈ 0.538. Subtracting the overlap is the whole point of the addition rule.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#2563eb;letter-spacing:1px">CHALLENGE 2 · MUTUALLY EXCLUSIVE</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">No overlap to subtract</div>
<div style="color:#4a5578;margin-top:6px">Roll one die. Find P(roll a 2 OR a 5). Are these events mutually exclusive? What happens to the addition rule?</div>
</div>

In [3]:
p2, p5 = 1/6, 1/6
print(f"P(2 or 5) = P(2) + P(5) = {p2:.3f} + {p5:.3f} = {p2+p5:.3f}")
print("P(2 AND 5) = 0 on a single die -> mutually exclusive, so no overlap term")

P(2 or 5) = P(2) + P(5) = 0.167 + 0.167 = 0.333
P(2 AND 5) = 0 on a single die -> mutually exclusive, so no overlap term


**Answer:** a single die cannot show both a 2 and a 5, so the events are mutually exclusive: P(2 and 5) = 0. The addition rule loses its overlap term and becomes plain addition, P(2 or 5) = 1/6 + 1/6 = 1/3.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#059669;letter-spacing:1px">CHALLENGE 3 · MULTIPLICATION (INDEPENDENT)</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Multiply along the AND</div>
<div style="color:#4a5578;margin-top:6px">Flip a fair coin and roll a die. Find P(heads AND a six). Are they independent? Verify by simulation.</div>
</div>

In [4]:
p_h, p_6 = 1/2, 1/6
p_and = p_h * p_6
print(f"P(heads AND six) = P(heads) x P(six) = {p_h} x {p_6:.3f} = {p_and:.3f}")
flips = rng.integers(0,2,300_000); rolls = rng.integers(1,7,300_000)
emp = np.mean((flips==1) & (rolls==6))
print(f"empirical (300k)                    = {emp:.3f}")

P(heads AND six) = P(heads) x P(six) = 0.5 x 0.167 = 0.083
empirical (300k)                    = 0.084


**Answer:** the coin and the die do not influence each other, so they are independent and the probabilities multiply: P(heads and six) = 1/2 x 1/6 = 1/12 ≈ 0.083, confirmed by simulation. For independent events, AND means multiply.

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#db2777;letter-spacing:1px">CHALLENGE 4 · TEST INDEPENDENCE</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Same or different?</div>
<div style="color:#4a5578;margin-top:6px">Two events have P(A)=0.5, P(B)=0.4. In case (i) P(A and B)=0.2; in case (ii) P(A and B)=0. For each, decide whether they are independent, mutually exclusive, or neither.</div>
</div>

In [5]:
pA, pB = 0.5, 0.4
for label, pAB in [("case (i)",0.2),("case (ii)",0.0)]:
    indep = np.isclose(pAB, pA*pB)
    mutex = np.isclose(pAB, 0.0)
    print(f"{label}: P(A and B)={pAB},  P(A)xP(B)={pA*pB:.2f}  ->  independent={indep}, mutually exclusive={mutex}")

case (i): P(A and B)=0.2,  P(A)xP(B)=0.20  ->  independent=True, mutually exclusive=False
case (ii): P(A and B)=0.0,  P(A)xP(B)=0.20  ->  independent=False, mutually exclusive=True


**Answer:** case (i) has P(A and B) = 0.2 = 0.5 x 0.4, so the events are **independent** (and clearly not mutually exclusive, they overlap). Case (ii) has P(A and B) = 0, so they are **mutually exclusive**, and since 0 ≠ 0.20 they are *not* independent. Same two events, opposite relationships, decided only by the test P(A and B) = P(A)xP(B).

<div style="background:#eef2ff;border-left:5px solid #4f46e5;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#d97706;letter-spacing:1px">CHALLENGE 5 · COMBINING RULES</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">At least one</div>
<div style="color:#4a5578;margin-top:6px">A free-throw shooter makes 80% of shots. Assuming independent shots, find P(at least one miss in 5 attempts) using the complement.</div>
</div>

In [6]:
p_make = 0.8
p_all_make = p_make**5
p_at_least_one_miss = 1 - p_all_make
print(f"P(make all 5)          = 0.8^5 = {p_all_make:.3f}")
print(f"P(at least one miss)   = 1 - that = {p_at_least_one_miss:.3f}")
shots = rng.random((300_000,5)) < 0.8
emp = np.mean(~shots.all(axis=1))
print(f"empirical (300k)       = {emp:.3f}")

P(make all 5)          = 0.8^5 = 0.328
P(at least one miss)   = 1 - that = 0.672
empirical (300k)       = 0.672


**Answer:** "at least one miss" is the complement of "makes all five". Made shots are independent, so P(all five) = 0.8^5 ≈ 0.328 by the multiplication rule, and P(at least one miss) = 1 - 0.328 ≈ 0.672. Even an 80% shooter misses at least once about two-thirds of the time over five attempts, the complement makes that easy to see.

---
<div style="background:#ffffff;border:1px solid #e6e9f2;border-radius:16px;padding:22px 26px;font-family:Inter,sans-serif">
<div style="font-size:19px;font-weight:800;color:#1a2138">🎉 Nicely done</div>
<div style="color:#4a5578;line-height:1.8;margin-top:8px">You added with and without overlap, multiplied across independent events, separated mutually exclusive from independent with the test, and combined rules for an "at least one" question. These four moves cover most everyday probability calculations.</div>
</div>

<div style="text-align:center;color:#8b94b3;font-size:12px;margin-top:16px">Statistics, Data Science and AI: A Visual Handbook · © 2026 John Fisher</div>